crates/polars-core/src/chunked_array/ops/filter.rs

```diff
@@ -14,6 +14,35 @@ macro_rules! check_filter_len {
     }};
 }
 
+/// Checks if a lambda expression is a length comparison
```

Comment:
My only comment is that I don't think this part should be here, this kind of optimization should be done in Scala where it is applied once for all occurrences, and creates a direct replacement that is very fast(here we still have to attempt the optimization on every executor for every partition)